test/rules/dangling_library_doc_comments_test.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(DanglingLibraryDocCommentsTest);
+  });
+}
+
+@reflectiveTest
+class DanglingLibraryDocCommentsTest extends LintRuleTest {
+  @override
+  String get lintRule => 'dangling_library_doc_comments';
+
+  test_doc_comment_above_declaration() async {
```

Comment:
These names don't following the convention for test names. See https://github.com/dart-lang/sdk/blob/main/pkg/analyzer/doc/implementation/tests.md#test-names.

Reply:
Oops, I copied `discared_futures_test`. Fixed. I kept `docComment_` in each name but let me know if you don't like it. I think it improves readability.